src/App.jsx

```diff
@@ -1,247 +1,309 @@
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
-              >
-                Max
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
 
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
+                {inputState.length > 0 && (
+                  <span className="icon is-right">
+                    {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                    <button
+                      data-cy="ClearButton"
+                      type="button"
+                      className="delete"
+                      onClick={() => setInputState('')}
+                    />
+                  </span>
+                )}
+
+              </p>
+            </div>
+
+            <div className="panel-block is-flex-wrap-wrap">
+              <a
+                href="#/"
+                data-cy="AllCategories"
+                className={cn('button', 'is-success', 'mr-6', {
+                  'is-outlined': categoriesList.length,
+                })}
+                onClick={() => setCategoriesList([])}
               >
-                Anna
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                3
-              </td>
-
-              <td data-cy="ProductName">iPhone</td>
-              <td data-cy="ProductCategory">💻 - Electronics</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
+                All
+              </a>
+
+              {categoriesFromServer.map(({ id, title }) => (
+                <a
+                  key={id}
+                  data-cy="Category"
+                  className={cn('button', 'mr-2', 'my-1', {
+                    'is-info': categoriesList.includes(title),
+                  })}
+                  href="#/"
+                  onClick={() => {
+                    if (!categoriesList.includes(title)) {
+                      setCategoriesList(prevState => ([
+                        ...prevState,
+                        title,
+                      ]));
+                    } else {
+                      setCategoriesList((prevState) => {
+                        const copy = [...prevState];
+
+                        copy.splice(copy.indexOf(title), 1);
+
+                        return copy;
+                      });
+                    }
+                  }}
```

Comment:
such functionality should be moved in separate function before the `return` statement